### PR TITLE
Separate out JS generator type; Add blockly categories

### PR DESCRIPTION
**Pull request by Casper-Oakley**
Cleaned up the blockly toolbox a little to encapsulate different sections in different folders. This doesn't add any new blocks, and doesn't change any block shapes or descriptions or anything, just moves them to categories (in preparation for more to come).

I've also separated out the JS generator type which had been defined in multiple places into a single type.

### CODE REVIEW

File: src/BlocklyInterface/robotblocks/actuators/DcMotor.ts [hunk 1]
```diff
@@ -43,19 +43,18 @@ export function addDcMotorBlock() {
...
         JavaScript.ORDER_ATOMIC
       );
 
       // convert direction to power sign +/-
-      var isForward = dropdown_direction === "FORWARD";
-      let sign = isForward ? "1" : "-1";
-      var code = `setDcMotorPower(${number_port}, ${sign} * (${value_power}));\n`;
-      return code;
+      const isForward = dropdownDirection === "FORWARD";
+      const sign = isForward ? "1" : "-1";
```
Comment (sar-k) [inline src/BlocklyInterface/robotblocks/actuators/DcMotor.ts:56]: should sign not be a number instead of string

Comment (Casper-Oakley) [reply src/BlocklyInterface/robotblocks/actuators/DcMotor.ts:56]: It is a number, though it's going in a string - the output of the func is a string of a js function:
`"1" : "-1"` -> ```setDcMotorPower(1, 1 * (0.1));```

versus if it were attempting to output a string:
`"\"1\"" : "\"-1\""` -> ```setDcMotorPower(1, "1" * (0.1));```

**Comment by sar-k** [comment]
@Casper-Oakley looks good except the unit tests are failing

**Comment by Casper-Oakley** [comment]
> @Casper-Oakley looks good except the unit tests are failing

yep, you're completely right, my bad! I'll push an update with the tests in a sec :)

File: src/RobotSimulator/robotSimulatorSlice.ts [hunk 2]
```diff
@@ -1,10 +1,28 @@
...
  */
 export const robotSimulatorSlice = createSlice({
-  initialState: {},
+  initialState: {
+    motors: {},
+  } as IRobotSimulatorState,
   name: "simulator",
-  reducers: {},
+  reducers: {
+    setPower(state, action: PayloadAction<{ channel: number; power: number }>) {
```
Comment (sar-k) [inline src/RobotSimulator/robotSimulatorSlice.ts]: nit: You may want to rename the method as setMotorPower to align with the getter

**Pull request merged**